src/node/markdown/plugins/containers.ts

```diff
@@ -15,31 +16,23 @@ export const containerPlugin = (md: MarkdownIt) => {
     })
 }
 
-type ContainerArgs = [
-  typeof container,
-  string,
-  {
-    render(tokens: Token[], idx: number): string
-  }
-]
+type ContainerArgs = [typeof container, string, { render: RenderRule }]
 
 function createContainer(klass: string, defaultTitle: string): ContainerArgs {
   return [
     container,
     klass,
     {
-      render(tokens, idx) {
+      render(tokens, idx, options, env) {
+        const md = new MarkdownIt().set(options)
```

Comment:
I think we can passing the `md` instance into the `createContainer` function, instead of creating a new one here. It will create a new instance for every container.

---

Oh I remember it. The fifth arg of the render function is the renderer. You can also consider using it instead

```ts
render(tokens, idx, options, env, slf) {
  slf.renderInline()
}
```

Reply:
It needs a token array to be passed, instead of text I guess.